src/redux/books/books.js

```diff
@@ -0,0 +1,29 @@
+const REMOVE = 'bookstore/books/REMOVE';
+const CREATE = 'bookstore/books/CREATE';
+
+const defaultState = [];
+
+// define an action creator for adding books
+export const addBook = (payload) => ({
+  type: CREATE,
+  payload,
+});
+
+// define an action creator for removing books
+export const removeBook = (payload) => ({
+  type: REMOVE,
+  payload,
+});
+
+export default function books(state = defaultState, action) {
+  switch (action.type) {
+    case CREATE:
+      return [...state, action.payload];
+
+    case REMOVE:
+      return [...state, action.payload];
```

Comment:
Great job @AlucardSanin 🤝

- [ ] Kindly make sure that in case action. type is REMOVE_BOOK the function uses the ES6 filter() method to create a new array, which will not contain the book you want to remove from the store (filter by the id key - i.e.:
return state.filter(book => book.id !== id).

Reply:
Hi there @1GORDON Thanks a lot for your review! 

Using the spread operator [...] you can do that also you don't need to implement the filter, of course, it will do the work as we learn in the freeCodeCamp exercises early in this module.

The spread operator was added to JavaScript in ES6 (ES2015), just like the rest parameters, which have the same syntax: three magic dots […]

Reply:
@AlucardSanin because you are not adding to your state array, in this case you have to filter the state to remove the item with the payload or find and splice the item which matches the payload passed, hope this makes sense, kindly fix this :clap:

Reply:
- [ ] as I have already mentioned in the comment, kindly use filter to remove the item in your state that matches the payload passed

Reply:
@VICTOR-CHUKWU Completely right, forgot to, it's a bit difficult to notice that when you're not running the store in the app, thanks!